gpflow/expectations.py

```diff
@@ -13,75 +13,214 @@
 # limitations under the License.from __future__ import print_function
 
 import functools
+import warnings
 import itertools as it
+
+import numpy as np
 import tensorflow as tf
 
 from . import kernels, mean_functions, settings
-from .probability_distributions import Gaussian, DiagonalGaussian, MarkovGaussian
-from .expectations_quadrature import dispatch, quadrature_fallback
 from .features import InducingFeature, InducingPoints
 from .decors import params_as_tensors_for
+from .quadrature import mvnquad
+from .probability_distributions import Gaussian, DiagonalGaussian, MarkovGaussian
+
+from multipledispatch import dispatch
+from functools import partial
+
+# By default multipledispatch uses a global namespace in multipledispatch.core.global_namespace.
+# We define our own GPflow namespace to avoid any conflict which may arise.
+gpflow_md_namespace = dict()
+dispatch = partial(dispatch, namespace=gpflow_md_namespace)
+
+
+# ========================== QUADRATURE EXPECTATIONS ==========================
+
+def quadrature_expectation(p, obj1, obj2=None, quad_points=None):
```

Comment:
much better now that user can control the number of gauss_hermite points.
Should we rename it though to the standard name that we've been using so far? `num_gauss_hermite_points`